src/Altinn.App.Core/Helpers/Serialization/ModelDeserializer.cs

```diff
@@ -9,6 +9,9 @@ namespace Altinn.App.Core.Helpers.Serialization;
 /// <summary>
 /// Represents logic to deserialize a stream of data to an instance of the given type
 /// </summary>
+// [Obsolete(
+//     "This class is deprecated and will be removed in a v9. Use Altinn.App.PlatformServices.Helpers.Serialization.ModelSerializationSerivce from dependency injection instead."
```

Comment:
It is not a super clean upgrade, so I'm not sure I'll actually deprecated this (thus the comment). I'm very open to suggestions on how to organise the code in `CachedInstanceDataAccessor`, `DataClient` and the serialisation logic.

Reply:
What method to use depends on how you want to select between json and xml.

If it is from storage, you need to use DeserialiseFromStorage, and pass the DataElement, so that we (in future) can look at AllowedContentTypes to select json or xml.

If the data comes from another source, you can use `DeserializeXml` or `DeserializeJson` or maybe use the (currently commented out) `DeserializeSingleFromRequest` that looks at a httpReqest to negotiate content type.